src/server/auth/server/api_server.go

```diff
@@ -1812,7 +1920,10 @@ func (a *apiServer) SetACLInTransaction(
 			}
 			aclMu.Lock()
 			defer aclMu.Unlock()
-			newACL.Entries[principal] = scope
+			// If the list includes multiple scopes for one principal, choose the highest
+			if scope > newACL.Entries[principal] {
```

Comment:
This makes SetACL deterministic if the same user is supplied multiple times with different scopes. This allows us to append an entry to the default ACL to grant the repo creator OWNER.

Reply:
This makes sense to me—maybe it would be helpful to note that this is for the new repo case (i.e. so that appending `<creater, OWNER>` works) in the comment?

Also, maybe this would be a good opportunity to get rid of the errgroup here and just do all of this serially? The thinking in making it parallel was that `canonicalizeSubject` can send an RPC to GitHub, but I know Go maps aren't thread-safe (using one here was always a mistake) and between that and GitHub not being popular anyway, it seems like the parallelism here might be net-negative value